### PR TITLE
Prevent Early Clan Units in Non-Clan Campaigns

### LINKED ISSUE #4975: [RFE] Keep Clan Mechs out of Unit Market in early IS games if Clan Purchasing is Off
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the RFE to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] Does this conform with our Content Policy linked above?
- [X] I've searched the GitHub tracker and haven't found a similar feature request listed

### RFE Details *

This is an improvement to an existing feature

### Brief Description *

Occasionally a clan mech will show up in the Unit Market, even when you're playing as IS in the 3020s with purchasing clan equipment turned off. Turning off clan purchasing should keep this from happening.

#4379 Involves adding clan mechs to the unit market for IS players after a certain date; it's the only other one I found related to this issue.

(Atma asked me to file this RFE based on conversation between her and Illiani in Discord))

### MekHQ Suite Version *

v0.50.0

### Attach Files

_No response_

### Final Checklist

- [X] I've checked to make sure that this RFE has not already been filed
- [X] I am requesting implementation of only one relevant set of features in this ticket for clarity and focus

### PR DESCRIPTION
**Pull request by IllianiCBT**
Added a check to prevent Clan units from appearing in the unit market for non-Clan campaigns, prior to the conclusion of the battle of Tukayyid.

### Closes #4975

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/market/unitMarket/AbstractUnitMarket.java [hunk 1]
```diff
@@ -191,8 +191,20 @@ public abstract void addOffers(final Campaign campaign, final int number, UnitMa
     public String addSingleUnit(final Campaign campaign, final UnitMarketType market,
             final int unitType, final MekSummary mekSummary,
             final int percent) {
+        final LocalDate INVASION_WAVE_FIVE = LocalDate.of(3051, 11, 1);
```
Comment (HammerGS) [inline MekHQ/src/mekhq/campaign/market/unitMarket/AbstractUnitMarket.java:194]: How about we turn this into a variable players can manage. Most of the Clan equipment starts to hit later per Merc 3055, After The Battle of Tukayyid in May of 3052.  

This isn't a thing for this PR but we should also be modelling that prior to this most houses (FRR and Kurita being the exception) weren't giving salvage rights to Mercs but exchange.

**Review by HammerGS — APPROVED** [review]
Approving but would like to look into the variable date.

File: MekHQ/src/mekhq/campaign/market/unitMarket/AbstractUnitMarket.java (hunk 1 above)
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/market/unitMarket/AbstractUnitMarket.java:194]: Happy to change this to after Tukayyid, but not interested in adding new campaign options until after CO IIC is merged as it just makes extra work for me to incorporate into CO IIC.

Comment (HammerGS) [reply MekHQ/src/mekhq/campaign/market/unitMarket/AbstractUnitMarket.java]: Works for me.

**Comment by IllianiCBT** [comment]
Switched cut-off date from the fifth wave to the Battle of Tukayyid, as per Hammer's feedback.

Description updated accordingly.

**Pull request merged**